docs: Canvas updates adding why and how describing data needs (#92375)
grafana/grafana: docs/sources/panels-visualizations/visualizations/canvas/index.md

# Canvas

Canvases combine the power of Grafana with the flexibility of custom elements.
They are extensible visualizations that allow you to add and arrange elements wherever you want within unstructured static and dynamic layouts.
This lets you design custom visualizations and overlay data in ways that aren't possible with standard Grafana visualizations, all within the Grafana UI.

{{< video-embed src="/static/img/docs/canvas-panel/canvas-beta-overview-9-2-0.mp4" max-width="750px" alt="Canvas beta overview" >}}

If you've used popular UI and web design tools, then designing canvases will feel very familiar.

> With all of these dynamic elements, there are few limits to what a canvas can display.

Reviewer: I know what you mean with this but to me it feels like we're gonna start listing the limits.. 😅

Reviewer 2: ```suggestion
With all of these dynamic elements, there's almost no limit to what a canvas can display.
```

{{< admonition type="note" >}}
We'd love your feedback on the canvas visualization. Please check out the [open Github issues](https://github.com/grafana/grafana/issues?page=1&q=is%3Aopen+is%3Aissue+label%3Aarea%2Fpanel%2Fcanvas) and [submit a new feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=type%2Ffeature-request,area%2Fpanel%2Fcanvas&title=Canvas:&projects=grafana-dataviz&template=1-feature_requests.md) as needed.
{{< /admonition >}}

## Supported data formats

> The canvas visualization is unique in that it doesn't have any specific data requirements. You can even start drawing and configuring visual elements without providing any data. However, any data you plan to consume should be accessible through supported Grafana data sources and structured in a way that ensures smooth integration with your custom elements.

Reviewer: _drawing_ ?

Reviewer 2: Better?
```suggestion
The canvas visualization is unique in that it doesn't have any specific data requirements. You can even start adding and configuring visual elements without providing any data. However, any data you plan to consume should be accessible through supported Grafana data sources and structured in a way that ensures smooth integration with your custom elements.
```

If your canvas is going to update in real time, your data should support refreshes at your desired intervals without degrading the user experience.

> You can provide data to the multiple _elements_ and _connections_, like text, colors, and background pattern images, available in the canvas visualization. These have different data requirements defined in the [Elements](#elements) and [Connections](#connections) sections.

Reviewer: > like text, colors, and background pattern images
Looks like we're referring to them as elements and connections? They're options. I'm having a hard time understing this entire section 😅

Reviewer 2: @adela-almasan does this make sense? @srperf - does this align with what you intended?
```suggestion
You can provide data to the multiple _elements_ and _connections_ available in the canvas visualization to set options like text, colors, and background pattern images. These have different data requirements defined in the [Elements](#elements) and [Connections](#connections) sections.
```

Reviewer: maybe something along these lines?

```suggestion
You can tie [Elements](#elements) and [Connections](#connections) to data through options like text, colors, and background pattern images, etc. available in the canvas visualization.
```
